.github/workflows/ci.yml

```diff
@@ -0,0 +1,90 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  qa:
+    name: QA
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toxenv:
+          - checkqa
+          - readme
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+      - name: Set PY
+        run: echo "::set-env name=PY::$(python -VV | sha256sum | cut -d' ' -f1)"
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Install tox
+        run: pip install tox
+      - name: Test ${{ matrix.toxenv }}
+        run: tox -e ${{ matrix.toxenv }}
+
+  cpython:
+    name: Test
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os:
+          - Ubuntu
+          - Windows
+        python-version:
+          - 2.7
+          - 3.5
+          - 3.6
+          - 3.7
+          - 3.8
+        pip-version:
+          - "20.0"
+          - "latest"
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: pip install tox
+      - name: Test pip ${{ matrix.pip-version }}
+        run: tox -e pip${{ matrix.pip-version }}-coverage
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1.0.6
+        with:
+          file: ./coverage.xml
+          name: ${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.pip-version }}
+
+  pypy:
```

Comment:
I don't think this deserves a separate job setup. It could be handled in the previous matrix.

Reply:
Yes, it could but it would take too many lines to run pypy jobs only with the latest pip version and ubuntu (which is done to reduce job numbers previously, still actual though) using exclude directive.

Reply:
That's why I decided to put it under a separate job

Reply:
I'm not convinced. Excludes may actually be way more flexible than you think. You shouldn't need to declare many of them.

Reply:
Could you send a diff with updated cpython job wich has 2 additional pypy jobs described below ?

- ubuntu + piplatest + pypy2
- ubuntu + piplatest + pypy3

Let see how it would look like.

Reply:
Done: https://github.com/jazzband/pip-tools/pull/1012#discussion_r419004600

Reply:
Thanks!  Addressed in 988dfbc.